### DISCUSSION TITLE
Form shows behind the parent form when showing through button

### DISCUSSION THREAD

**Discussion by sampathnarine** [dotnet/winforms#4893] [Q&A]
Hi, 

I have created the winforms sample. I have set form's visibility as true when mouse pointer enters the button. So the form opens and shown in the toplevel. Now click the button and again hover on the button. The form opens in behind the other form. I felt the reason is focus is in the button.  I feel the form which shows while mouse hover should be in toplevel even after click the button. 

Run the sample
Hover the button
Form opens in the toplevel
Click the button
Again hover the Button. Now the form will shown behind the parent form.

[Sample.zip](https://github.com/dotnet/winforms/files/6459630/Sample.zip)

**Comment by sampathnarine** [comment]
Hi,

Any update for this query.

Regards,
Sampath Narayanan.S

**Comment by weltkante** [comment]
This seems to be expected behavior from Windows, if you just toggle visibility you aren't changing the order of forms. You may want to call `BringToFront` or `Show` if you want to do more than just making it visible.

The reason the initial `Visible = true` assignment brings it to the front is because making it visible for the first time actually creates the form, and newly created forms are on the top. Subsequently you just hide it, so by actively clicking the button Windows brings this form to the front again; making the other form visible again later retains that updated window order.
